### PR TITLE
proxyserver: add stats endpoint for monitoring

### PR DESCRIPTION
**Pull request by AmarOk1412**
Just to be compatible with my monitor app :)

### CODE REVIEW

File: src/dht_proxy_server.cpp [hunk 1]
```diff
@@ -234,6 +234,50 @@ DhtProxyServer::getNodeInfo(const std::shared_ptr<restbed::Session>& session) co
...
+    int content_length = std::stoi(request->get_header("Content-Length", "0"));
+    session->fetch(content_length,
+        [this](const std::shared_ptr<restbed::Session> s, const restbed::Bytes& /*b*/) mutable
+        {
+            try {
+                if (dht_) {
+                    Json::Value result;
+
+                    auto stats = stats_;
+                    result["listenCount"] = static_cast<uint64_t>(stats.listenCount);
```
Comment (aberaud) [inline src/dht_proxy_server.cpp]: Make serverstats json-serialisable

**Comment by AmarOk1412** [comment]
Done :)

File: include/opendht/dht_proxy_server.h [hunk 2]
```diff
@@ -73,15 +77,46 @@ class OPENDHT_PUBLIC DhtProxyServer
...
+        NodeInfo nodeInfo;
 
         std::string toString() const {
             std::ostringstream ss;
             ss << "Listens: " << listenCount << " Puts: " << putCount << " PushListeners: " << pushListenersCount << std::endl;
             ss << "Requests: " << requestRate << " per second." << std::endl;
+            auto& ni = nodeInfo;
+            auto& ipv4 = ni.ipv4;
+            if (ipv4.table_depth > 1) {
+                unsigned long tot_nodes = 8 * std::exp2(ipv4.table_depth);
```
Comment (aberaud) [inline include/opendht/dht_proxy_server.h]: Add method getNetworkSizeEstimation to NodeStats to factorize computation

Comment (AmarOk1412) [reply include/opendht/dht_proxy_server.h]: Done

**Pull request merged**